### PR TITLE
Fixing warnings

### LINKED ISSUE #10: Include the sphinx "roles" used to improve PDK documentation to enable sharing between PDKs
The SkyWater PDK defines a bunch of "rst roles" in it's [sphinx `conf.py` file](https://github.com/google/skywater-pdk/blob/main/docs/conf.py). These include things like;
 * lib
 * cell
 * model

Which are used like this in the SkyWater PDK documentation;
```rst
:lib:`sky130_fd_io` - SKY130 IO and periphery cells (SkyWater Provided)
```
and like follows;
```rst
Libraries in the SKY130 PDK are named using the following scheme;

  :lib_process:`<Process name>` _ :lib_src:`<Library Source Abbreviation>` _ :lib_type:`<Library Type Abbreviation>` [_ :lib_name:`<Library Name>`]
```

Which results in the following colorization in the PDK documentation;
image

In the future, it would also be good if it resulted in these nodes also being clickable links to useful things.

This is useful for documenting all the PDKs which follow our naming process. We should move it into this repository and make it shared.

This is done with the following Python code;
```python
import re
from docutils.parsers.rst import directives, roles, nodes

LIB_REGEX = re.compile('sky130_(?P<lib_src>[^_\s]*)_(?P<lib_type>[^_\s]*)(_(?P<lib_name>[^_\s]*))?')
CELL_REGEX = re.compile('sky130_(?P<lib_src>[^_\s]*)_(?P<lib_type>[^_\s]*)(_(?P<lib_name>[^_\s]*))?__(?P<cell_name>[^\s]*)')

def lib_role(name, rawtext, text, lineno, inliner, options={}, content=[]):
    """Library name which gets colorized."""
    m = LIB_REGEX.match(text)
    if not m:
        msg = inliner.reporter.error("Malformed library name of "+repr(text), line=lineno)
        prb = inliner.problematic(rawtext, rawtext, msg)
        return [prb], [msg]
    app = inliner.document.settings.env.app

    #lib_process_role = roles.role('lib_src', inliner.language, lineno, inliner.reporter)
    #lib_src_role = roles.role('lib_src', inliner.language, lineno, inliner.reporter)
    #lib_type_role = roles.role('lib_src', inliner.language, lineno, inliner.reporter)
    #lib_name_role = roles.role('lib_src', inliner.language, lineno, inliner.reporter)
    lib_process = 'sky130'
    lib_src = m.group('lib_src')
    lib_type = m.group('lib_type')
    lib_name = m.group('lib_name')

    r = [
        nodes.inline(lib_process, lib_process, classes=['lib-process']),
        nodes.inline('_', '_', options=options),
        nodes.inline(lib_src, lib_src, classes=['lib-src']),
        nodes.inline('_', '_', options=options),
        nodes.inline(lib_type, lib_type, classes=['lib-type']),
    ]
    if lib_name:
        r.append(nodes.inline('_', '_', options=options))
        r.append(nodes.inline(lib_name, lib_name, classes=['lib-name']))

    return r, []


def cell_role(name, rawtext, text, lineno, inliner, options={}, content=[]):
    """Cell name which gets colorized."""
    m = CELL_REGEX.match(text)
    if not m:
        msg = inliner.reporter.error("Malformed cell name of "+repr(text), line=lineno)
        prb = inliner.problematic(rawtext, rawtext, msg)
        return [prb], [msg]
    app = inliner.document.settings.env.app

    #lib_process_role = roles.role('lib_src', inliner.language, lineno, inliner.reporter)
    #lib_src_role = roles.role('lib_src', inliner.language, lineno, inliner.reporter)
    #lib_type_role = roles.role('lib_src', inliner.language, lineno, inliner.reporter)
    #lib_name_role = roles.role('lib_src', inliner.language, lineno, inliner.reporter)
    lib_process = 'sky130'
    lib_src = m.group('lib_src')
    lib_type = m.group('lib_type')
    lib_name = m.group('lib_name')
    cell_name = m.group('cell_name')

    r = [
        nodes.inline(lib_process, lib_process, classes=['lib-process']),
        nodes.inline('_', '_', options=options),
        nodes.inline(lib_src, lib_src, classes=['lib-src']),
        nodes.inline('_', '_', options=options),
        nodes.inline(lib_type, lib_type, classes=['lib-type']),
    ]
    if lib_name:
        r.append(nodes.inline('_', '_', options=options))
        r.append(nodes.inline(lib_name, lib_name, classes=['lib-name']))
    r.append(nodes.inline('__', '__', options=options))
    r.append(nodes.inline(cell_name, cell_name, classes=['cell-name']))

    return r, []


def add_role(app, new_role_name):
    options = {
        'class': directives.class_option(new_role_name),
    }
    role = roles.CustomRole(new_role_name, roles.generic_custom_role, options, "")
    app.add_role(new_role_name, role)


def setup(app):
    app.add_css_file('extra.css')
    add_role(app, 'cell_name')
    add_role(app, 'lib_process')
    add_role(app, 'lib_src')
    add_role(app, 'lib_type')
    add_role(app, 'lib_name')
    add_role(app, 'drc_rule')
    add_role(app, 'drc_tag')
    add_role(app, 'drc_flag')
    add_role(app, 'layer')

    app.add_role('lib', lib_role)
    app.add_role('cell', cell_role)
    app.add_role('model', cell_role)
```

### LINKED ISSUE #20: Fix all the warnings produced when building sphinx documentation
Currently there are a lot of warnings produce when building the sphinx documentation.

These should all be cleaned up and then enable the [`sphinx.fail_on_warning`](https://docs.readthedocs.io/en/stable/config-file/v2.html#sphinx-fail-on-warning) option so that new warnings are not introduced.

### LINKED ISSUE #22: Add documentation about the naming scheme
How things are named in the PDK should be included in the documentation (by including the common documentation from https://github.com/google/open-source-pdks).

See related issue @ https://github.com/google/open-source-pdks/issues/7

### PR DESCRIPTION
**Pull request by mohanad0mohamed**
Fixes https://github.com/google/gf180mcu-pdk/issues/20
Fixes https://github.com/google/gf180mcu-pdk/issues/22
Fixes #10 

Added the `extra.css` and `ext.py` (part of `conf.py`) as suggested here https://github.com/google/gf180mcu-pdk/pull/39#pullrequestreview-1059568565

- [x] Built locally successfully

### CODE REVIEW

File: docs/_static/extra.css [hunk 1]
```diff
@@ -0,0 +1,53 @@
+img.align-center {
```
Comment (mithro) [inline docs/_static/extra.css:19]: Missing license header.

File: docs/ext.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
+from docutils.parsers.rst import directives, roles, nodes
```
Comment (mithro) [inline docs/ext.py]: Missing license header.

**Review by mithro — APPROVED** [review]
Two minor comments and then this can be merged.

File: docs/ext.py (hunk 2 above)
Comment (mithro) [inline docs/ext.py]: We should name this file something like `sphinx_pdk_roles.py` -- see https://github.com/google/gf180mcu-pdk/pull/39/files#r938371142

Comment (mohanad0mohamed) [reply docs/ext.py]: Done here and I will change the name in `gf180mcu-pdk` repository also.

Comment (mohanad0mohamed) [reply docs/ext.py]: Done

File: docs/_static/extra.css (hunk 1 above)
Comment (mohanad0mohamed) [reply docs/_static/extra.css:19]: Done

**Comment by mohanad0mohamed** [comment]
@mithro I have made the requested changes. Also, I have fixed the warnings of this repository also.

**Review by mithro — APPROVED** [review]
LGTM.

**Pull request merged**